### PR TITLE
[nodeinfo] keep updating broadcast list

### PR DESCRIPTION
**Pull request by envestcc**
# Description
- use broadcastList instead of delegates to avoid ambiguity
- remove height dependent feature. Because node in old version only output some warn level message for unknown msg type, it's unnecessary to add that.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: chainservice/builder.go [hunk 1]
```diff
@@ -385,9 +384,23 @@ func (builder *Builder) buildNodeInfoManager() error {
...
-	dm := nodeinfo.NewInfoManager(&builder.cfg.NodeInfo, cs.p2pAgent, cs.chain, builder.cfg.Chain.ProducerPrivateKey(), func(ctx context.Context) (state.CandidateList, error) {
-		return stk.ActiveCandidates(ctx, cs.factory, 0)
+
+	dm := nodeinfo.NewInfoManager(&builder.cfg.NodeInfo, cs.p2pAgent, builder.cfg.Chain.ProducerPrivateKey(), func() []string {
+		whiteList := []string{}
+		candidates, err := stk.ActiveCandidates(context.Background(), cs.factory, 0)
+		if err != nil {
+			log.L().Error("failed to get active candidates", zap.Error(errors.WithStack(err)))
+		} else {
+			whiteList = make([]string, len(candidates))
```
Comment (dustinxie) [inline chainservice/builder.go:394]: `whitelist` already defined, and is empty []string at L389, why make again? i think you want to allocate the exact size, then this maybe be better?
```
if err != nil {
	log.L().xxx
	return nil
}

whiteList = make([]string, len(candidates))
...
return whiteList
```

File: chainservice/builder.go [hunk 2]
```diff
@@ -385,9 +384,23 @@ func (builder *Builder) buildNodeInfoManager() error {
...
+			for i := range whiteList {
+				whiteList[i] = candidates[i].Address
+			}
+		}
+		return whiteList
 	})
+	if err := cs.chain.AddSubscriber(dm); err != nil {
```
Comment (dustinxie) [inline chainservice/builder.go:401]: i think no need to use subscriber, see comments below

File: nodeinfo/config.go [hunk 3]
```diff
@@ -11,12 +11,14 @@ import "time"
 type Config struct {
 	EnableBroadcastNodeInfo   bool          `yaml:"enableBroadcastNodeInfo"`
 	BroadcastNodeInfoInterval time.Duration `yaml:"broadcastNodeInfoInterval"`
+	WhiteListTTL              time.Duration `yaml:"whiteListTTL"`
```
Comment (dustinxie) [inline nodeinfo/config.go:14]: `BroadcastListTTL`

File: nodeinfo/manager.go [hunk 4]
```diff
@@ -81,44 +75,38 @@ func init() {
 }
 
 // NewInfoManager new info manager
-func NewInfoManager(cfg *Config, t transmitter, ch chain, privKey crypto.PrivateKey, getDelegatesHandler delegatesGetFunc) *InfoManager {
+func NewInfoManager(cfg *Config, t transmitter, privKey crypto.PrivateKey, getWhiteListFunc getWhiteListFunc) *InfoManager {
```
Comment (dustinxie) [inline nodeinfo/manager.go:78]: i think we can keep the `ch chain`, see comment above regarding subscriber

File: nodeinfo/manager.go [hunk 5]
```diff
@@ -161,8 +149,8 @@ func (dm *InfoManager) updateNode(node *Info) {
 	_nodeInfoHeightGauge.WithLabelValues(addr, node.Version).Set(float64(node.Height))
 }
 
-// GetNodeByAddr get node info by address
-func (dm *InfoManager) GetNodeByAddr(addr string) (Info, bool) {
+// GetInfo get node info by address
+func (dm *InfoManager) GetInfo(addr string) (Info, bool) {
```
Comment (dustinxie) [inline nodeinfo/manager.go:153]: `GetNodeInfo`

File: nodeinfo/manager.go [hunk 6]
```diff
@@ -231,25 +225,14 @@ func (dm *InfoManager) genNodeInfoMsg() (*iotextypes.NodeInfo, error) {
...
 }
 
-func (dm *InfoManager) isDelegate() bool {
-	if !dm.delegateCache {
-		if err := dm.updateDelegateCache(); err != nil {
-			log.L().Error("nodeinfo manager update delegate cache failed", zap.Error(err))
-		}
-	}
-	return dm.delegateCache
+func (dm *InfoManager) inWhiteList() bool {
```
Comment (dustinxie) [inline nodeinfo/manager.go:228]: `inBroadcastList`

File: nodeinfo/manager.go [hunk 7]
```diff
@@ -231,25 +225,14 @@ func (dm *InfoManager) genNodeInfoMsg() (*iotextypes.NodeInfo, error) {
...
+	return slices.Contains(dm.whiteList.Load().([]string), dm.address)
 }
 
-func (dm *InfoManager) updateDelegateCache() error {
-	candList, err := dm.getDelegates(context.Background())
-	if err != nil {
-		return err
+func (dm *InfoManager) updateWhiteList() {
```
Comment (dustinxie) [inline nodeinfo/manager.go:232]: `updateBroadcastList`

File: nodeinfo/config.go [hunk 8]
```diff
@@ -11,12 +11,14 @@ import "time"
 type Config struct {
 	EnableBroadcastNodeInfo   bool          `yaml:"enableBroadcastNodeInfo"`
 	BroadcastNodeInfoInterval time.Duration `yaml:"broadcastNodeInfoInterval"`
+	BroadcastListTTL          time.Duration `yaml:"broadcastListTTL"`
```
Comment (dustinxie) [inline nodeinfo/config.go:14]: nit: `BroadcastListInterval`?

**Review by dustinxie — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

**Pull request merged**